example/server.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/domhoward14/swagno"
+	"github.com/domhoward14/swagno/components/endpoint"
+	"github.com/domhoward14/swagno/components/mime"
+	"github.com/domhoward14/swagno/components/parameter"
+	"github.com/domhoward14/swagno/example/models"
+	"github.com/domhoward14/swagno/http/response"
+
+	"github.com/go-swagno/swagno-http/swagger"
+)
+
+func main() {
+	sw := swagno.New(swagno.Config{Title: "Testing API", Version: "v1.0.0"})
+	desc := "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed id malesuada lorem, et fermentum sapien. Vivamus non pharetra risus, in efficitur leo. Suspendisse sed metus sit amet mi laoreet imperdiet. Donec aliquam eros eu blandit feugiat. Quisque scelerisque justo ac vehicula bibendum. Fusce suscipit arcu nisl, eu maximus odio consequat quis. Curabitur fermentum eleifend tellus, lobortis hendrerit velit varius vitae."
+
+	endpoints := []*endpoint.EndPoint{
+		endpoint.New(
+			endpoint.WithMethod(endpoint.GET),
+			endpoint.WithPath("/product"),
+			endpoint.WithTags("product"),
+			endpoint.WithSuccessfulReturns([]response.Info{models.UnsuccessfulResponse{}}),
+			endpoint.WithErrors([]response.Info{models.EmptySuccessfulResponse{}}),
+			endpoint.WithDescription(desc),
+			endpoint.WithProduce([]mime.MIME{mime.JSON, mime.XML}),
+			endpoint.WithConsume([]mime.MIME{mime.JSON}),
+			endpoint.WithSummary("this is a test summary"),
+		),
+		endpoint.New(
+			endpoint.WithMethod(endpoint.GET),
+			endpoint.WithPath("/product/{id}"),
+			endpoint.WithTags("product"),
+			endpoint.WithParams(parameter.IntParam("id", parameter.WithIn(parameter.Path), parameter.WithRequired())),
+			endpoint.WithSuccessfulReturns([]response.Info{models.SuccessfulResponse{}}),
+			endpoint.WithErrors([]response.Info{models.UnsuccessfulResponse{}}),
+			endpoint.WithProduce([]mime.MIME{mime.JSON, mime.XML}),
+		),
+		endpoint.New(
+			endpoint.WithMethod(endpoint.GET),
+			endpoint.WithPath("/product/{id}/detail"),
+			endpoint.WithTags("product"),
+			endpoint.WithParams(parameter.IntParam("id", parameter.WithIn(parameter.Path), parameter.WithRequired())),
+			endpoint.WithSuccessfulReturns([]response.Info{models.SuccessfulResponse{}}),
+			endpoint.WithErrors([]response.Info{models.UnsuccessfulResponse{}}),
+			endpoint.WithProduce([]mime.MIME{mime.JSON, mime.XML}),
+		),
+		endpoint.New(
+			endpoint.WithMethod(endpoint.POST),
+			endpoint.WithPath("/product"),
+			endpoint.WithTags("product"),
+			endpoint.WithBody(models.ProductPost{}),
+			endpoint.WithSuccessfulReturns([]response.Info{models.SuccessfulResponse{}}),
+			endpoint.WithErrors([]response.Info{models.UnsuccessfulResponse{}}),
+			endpoint.WithProduce([]mime.MIME{mime.JSON, mime.XML}),
+		),
+	}
+
+	// TODO make support for popular http server libs so that the creation of endpoints and handlers can happen in one funciton
```

Comment:
How can we achieve this? Because all web frameworks accept different types of `HandlerFunc`. For example: `gin.HandlerFunc`, `func(c *fiber.Ctx) error` etc. Thats why I created different packages for [gin](https://github.com/go-swagno/swagno-gin) and [fiber](https://github.com/go-swagno/swagno-fiber).

If we can not achieve this at the moment, we should add fiber and gin examples additionally I guess 😄

Reply:
I know fiber has adaptor for accepting `http.HandlerFunc` but I am not sure all frameworks also have adaptor-like feature

Reply:
Yeah, I was going to come back to this one. I want to research this one a little more. For now since we're merging back to the original repo I'll revert the gin and fiber examples like you suggested.